src/main/java/ua/edu/internship/interview/web/controller/UserQuestionController.java

```diff
@@ -0,0 +1,82 @@
+import java.util.List;
+
+@Tag(name = "User Skills and Questions", description = "User skills and questions management endpoints")
+@RestController
+@RequestMapping("/users/{userId}/questions")
+@RequiredArgsConstructor
+public class UserQuestionController {
+    private final UserQuestionService service;
+
+    @GetMapping
+    @Operation(summary = "Get user questions",
+            description = "Retrieves a list of all questions for a specific user.")
+    @ApiResponse(responseCode = "200", content = @Content(array = @ArraySchema(schema = @Schema(implementation = UserQuestionDto.class))))
+    @ApiResponse(responseCode = "404", description = "User not found")
+    public List<UserQuestionDto> getUserQuestions(@PathVariable String userId) {
+        return service.getUserQuestions(userId);
+    }
+
+    @PostMapping
+    @Operation(summary = "Create a new user question",
+            description = "Creates a new question for a specific user.")
+    @ApiResponse(responseCode = "201", content = @Content(schema = @Schema(implementation = UserQuestionDto.class)))
+    @ApiResponse(responseCode = "400", description = "Invalid input")
+    @ResponseStatus(HttpStatus.CREATED)
+    public UserQuestionDto createUserQuestion(@PathVariable String userId, @RequestBody @Valid UserQuestionCreateDto dto) {
+        return service.createUserQuestion(userId, dto);
+    }
+
+    @PutMapping("/{questionId}")
+    @Operation(summary = "Update an existing user question",
+            description = "Updates an existing user question.")
+    @ApiResponse(responseCode = "200", content = @Content(schema = @Schema(implementation = UserQuestionDto.class)))
+    @ApiResponse(responseCode = "400", description = "Invalid input")
+    @ApiResponse(responseCode = "404", description = "User or question not found")
+    public UserQuestionDto updateUserQuestion(@PathVariable String userId, @PathVariable String questionId,
+                                              @RequestBody UserQuestionUpdateDto dto) {
+        return service.updateUserQuestion(userId, questionId, dto);
+    }
+
+    @DeleteMapping("/{questionId}")
+    @Operation(summary = "Delete a user question",
+            description = "Deletes a specific question for a user.")
+    @ApiResponse(responseCode = "204", description = "Question deleted successfully")
+    @ApiResponse(responseCode = "404", description = "User or question not found")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteUserQuestion(@PathVariable String userId, @PathVariable String questionId) {
+        service.deleteUserQuestion(userId, questionId);
+    }
+
+    @GetMapping("/skill/{skillId}")
+    @Operation(summary = "Get user questions by skill",
+            description = "Retrieves questions related to a specific skill for a user.")
+    @ApiResponse(responseCode = "200", content = @Content(array = @ArraySchema(schema = @Schema(implementation = UserQuestionDto.class))))
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 120 characters (found 138).
<!-- __reviewdog__:ChBlMzFkNjUxYjlmMThiN2Q2EgpjaGVja3N0eWxl -->